src/socket/address.cr

```diff
@@ -75,6 +103,36 @@ class Socket
       end
     end
 
+    # Parses a `Socket::IPAddress` from an URI.
+    #
+    # It expects the URI to include `<scheme>://<host>:<port>` where `scheme` as
+    # well as any additional URI components (such as `path` or `query`) are ignored.
+    #
+    # `host` must be an IP address (v4 or v6), otherwise `Socket::Error` will be
+    # raised. Domain names will not be resolved.
+    #
+    # ```
+    # Socket::IPAddress.parse("tcp://127.0.0.1:8080") # => Socket::IPAddress.new("127.0.0.1", 8080)
+    # Socket::IPAddress.parse("udp://[::1]:8080")     # => Socket::IPAddress.new("::1", 8080)
+    # ```
+    def self.parse(uri : URI) : IPAddress
```

Comment:
Somehow I'm ok with `T.parse: URI -> T` 😹 , but not with Hash.

Reply:
URI is essentially a structured string...